core/src/main/scala/iliad/gl/glInterpreter.scala

```diff
@@ -0,0 +1,158 @@
+      ptr.get(arr, 0, num)
+      arr.toList
+  }
+
+  def apply[A](gl: GL[A]): Reader[GLES30Library, A] = gl match {
+
+    case GLGetError => Reader(_.glGetError())
+
+    case GLGetShaderiv(shader, pname) =>
+      Reader { lib =>
+        val ptr = Buffer.capacity[Int](1)
+        lib.glGetShaderiv(shader, pname.value, ptr)
+        ptr.get()
+      }
+
+    case GLGetShaderInfoLog(shader, maxLength) =>
+      Reader { lib =>
+        val lenPtr = Buffer.capacity[Int](1)
+        val logPtr = Buffer.capacity[Byte](maxLength)
+        lib.glGetShaderInfoLog(shader, maxLength, lenPtr, logPtr)
+        val len = lenPtr.get()
+        val arr = new Array[Byte](len)
+        logPtr.get(arr, 0, len)
+        new String(arr)
+      }
+
+    case GLCreateShader(t) => Reader(_.glCreateShader(t.value))
+    case GLShaderSource(shader, sources) =>
+      Reader(
+          _.glShaderSource(shader,
+                           sources.size,
+                           sources.toArray,
+                           sources.map(_.length).toArray))
+    case GLCompileShader(shader) => Reader(_.glCompileShader(shader))
+    case GLCreateProgram => Reader(_.glCreateProgram())
+    case GLAttachShader(program, shader) =>
+      Reader(_.glAttachShader(program, shader))
+    case GLLinkProgram(program) => Reader(_.glLinkProgram(program))
+    case GLGetAttribLocation(program, name) =>
+      Reader(_.glGetAttribLocation(program, name))
+
+    case GLGenBuffers(number) => genObject(_.glGenBuffers, number)
+    case GLBindBuffer(target, buffer) =>
+      Reader(_.glBindBuffer(target.value, buffer))
+    case GLBufferData(target, size, data, usage) =>
+      Reader(_.glBufferData(target.value, size, data, usage.value))
+    case GLBufferSubData(target, offset, size, data) =>
+      Reader(_.glBufferSubData(target.value, offset, size, data))
+    case GLCopyBufferSubData(read, write, readOffset, writeOffset, size) =>
+      Reader(_.glCopyBufferSubData(
+              read.value, write.value, readOffset, writeOffset, size))
+
+    case GLBindFramebuffer(target, framebuffer) =>
+      Reader(_.glBindFramebuffer(target.value, framebuffer))
+    case GLEnable(cap) => Reader(_.glEnable(cap.value))
+    case GLDisable(cap) => Reader(_.glDisable(cap.value))
+    case GLColorMask(r, g, b, a) => Reader(_.glColorMask(r, g, b, a))
+    case GLUseProgram(program) => Reader(_.glUseProgram(program))
+    case GLEnableVertexAttribArray(location) =>
+      Reader(_.glEnableVertexAttribArray(location))
+    case GLVertexAttribPointer(
+        location, size, t, normalized, stride, offset) =>
+      Reader(
+          _.glVertexAttribPointer(
+              location, size, t.value, normalized, stride, offset))
+    case GLDrawElements(mode, count, t, offset) =>
+      Reader(_.glDrawElements(mode.value, count, t.value, offset))
+    case GLClear(bitMask) => Reader(_.glClear(bitMask.value))
+  }
+}
+
+object GLLogInterpreter extends (GL.Interpreter[GL.LogEffect]) {
+
```

Comment:
That won't work.  If we add this in then the `WriterT` can't have a monad instance, since a `Monad[WriterT[F, L, V]]` requires a `Moniod[L]`   (for the Applicative's `pure` to work).